### PR TITLE
Fixed event-listener that updated coupon usage

### PR DESCRIPTION
**Pull request by mmoreram**
* Removed all useless tests (empty files just for testing a simple service existence)
* Updated tests to speed them all a little bit
* Added some coverage for this new feature
* Removed non-used testing code

### CODE REVIEW

File: src/Elcodi/Bundle/CartCouponBundle/Tests/Functional/EventListener/CartCouponRulesEventListenerTest.php [hunk 1]
```diff
@@ -109,81 +86,37 @@ public function testOnCartCouponApplyValidate(array $expressions, $couponsNumber
...
             [['true == false', '1 == 1', 'rule(0) and rule(1)'], 0],
             [['true == false', '1 == 1', 'rule(0) or rule(1)'], 1],
-            [['cart.getId() == 1', '1 == 2', 'rule(0) and rule(1)'], 0],
+            [['cart.getId() == 2', '1 == 2', 'rule(0) and rule(1)'], 0],
             [['cart.getId() == 3', 'cart.getId() != coupon.getId()', 'rule(0) and rule(1)'], 0],
             [
                 [
                     'few_products' => 'cart.getQuantity() < 5',
-                    'low_cost' => 'cart.getAmount() < 10',
+                    'low_cost'     => 'cart.getAmount() < 10',
```
Comment (EmanueleMinotto) [inline src/Elcodi/Bundle/CartCouponBundle/Tests/Functional/EventListener/CartCouponRulesEventListenerTest.php]: Should we add this rule to the `.php_cs` configuration?

File: src/Elcodi/Bundle/CartCouponBundle/Tests/Functional/EventListener/CartCouponStackableEventListenerTest.php [hunk 2]
```diff
@@ -32,16 +32,21 @@ class CartCouponStackableEventListenerTest extends WebTestCase
...
     /**
      * @var Cart
+     *
+     * Cart
      */
     protected $cart;
 
     /**
      * @var CartCouponManager
+     *                        Cart Coupon manager
```
Comment (EmanueleMinotto) [inline src/Elcodi/Bundle/CartCouponBundle/Tests/Functional/EventListener/CartCouponStackableEventListenerTest.php]: Missing the right indentation.

Comment (mmoreram) [inline src/Elcodi/Bundle/CartCouponBundle/Tests/Functional/EventListener/CartCouponStackableEventListenerTest.php]: yes, my fault, fixing :dart:

File: src/Elcodi/Bundle/TestCommonBundle/Functional/WebTestCase.php [hunk 3]
```diff
@@ -238,9 +170,9 @@ protected function createSchema()
             '--quiet'          => true,
         ]));
 
-        $this->loadFixtures();
+        static::loadFixtures();
 
-        return $this;
+        return;
```
Comment (EmanueleMinotto) [inline src/Elcodi/Bundle/TestCommonBundle/Functional/WebTestCase.php]: A `return` here is no more required.

File: src/Elcodi/Bundle/TestCommonBundle/Functional/WebTestCase.php [hunk 4]
```diff
@@ -252,28 +184,30 @@ protected function createSchema()
...
+            static::$application->run(new ArrayInput([
+                'command'          => 'doctrine:fixtures:load',
+                '--no-interaction' => true,
+                '--fixtures'       => $formattedBundles,
+                '--quiet'          => true,
+            ]));
+        }
 
-        return $this;
+        return;
```
Comment (EmanueleMinotto) [inline src/Elcodi/Bundle/TestCommonBundle/Functional/WebTestCase.php]: Same as above.

**Comment by EmanueleMinotto** [comment]
Well done! :+1:
Should we consider https://github.com/matthiasnoback/SymfonyDependencyInjectionTest for future tests of services?

File: src/Elcodi/Bundle/TestCommonBundle/Functional/WebTestCase.php (hunk 3 above)
Comment (mmoreram) [inline src/Elcodi/Bundle/TestCommonBundle/Functional/WebTestCase.php]: It is, otherwise you would continue to the next line ;)

**Comment by mmoreram** [comment]
[Fixing]

Yes, we should consider it ;)

**Pull request merged**